modules/creating-cluster-with-aws-kms-key.adoc

```diff
@@ -0,0 +1,198 @@
+// Module included in the following assemblies:
+//
+// * monitoring/enabling-monitoring-for-user-defined-projects.adoc
+
+:_content-type: PROCEDURE
+[id="creating-cluster-with-aws-kms-key"]
+= Creating a ROSA cluster using a custom AWS KMS key
+
+You can create a {product-title} (ROSA) cluster with a customer-provided KMS key that is used to encrypt both node root volumes and persistent volumes for mounted EBS claims.
+
+.Procedure
+
+. Create a custom AWS customer-managed KMS key by running the following command:
++
+[source,terminal]
+----
+$ KMS_ARN=$(aws kms create-key --region $AWS_REGION --description 'Custom ROSA Encryption Key' --tags TagKey=red-hat,TagValue=true --query KeyMetadata.Arn --output text)
+----
++
+This command saves the Amazon Resource Name (ARN) output of this custom key for further steps.
+
+. Verify the KMS key has been created:
++
+[source,terminal]
+----
+$ echo $KMS_ARN
+----
+
+. Generate the necessary key policy to allow the ROSA STS roles to access the custom KMS key. Use the following command to populate a sample policy, or create your own.
+
++
+[source,terminal]
+----
+$ EBS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-openshift-cluster-csi-drivers-ebs-cloud-credential" | cut -c -64) 
+KUBE_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kube-controller-manager" | cut -c -64)
+KMS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kms-provider" | cut -c -64)
+CAPA_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-capa-controller-manager" | cut -c -64) <1>
```

Comment:
The `OPERATOR_ROLES_PREFIX` environment variable isn't set anywhere. My thoughts...

- Either make this something like the `<prefix>` that we reference in the "Creating Operator roles and policies" module
- Or, update the "Creating Operator roles and policies" module to include a step that saves `<prefix>` to the environment variable `OPERATOR_ROLES_PREFIX`